### PR TITLE
Home work 10

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/Task1.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	private static void anotherMethod() {
+		System.out.println(getCallerClassAndMethodName());
+	}
+
+	public static String getCallerClassAndMethodName() {
+		StackTraceElement[] stackTraceElement = new Exception().getStackTrace();
+		if (stackTraceElement.length < 3) {
+			return null;
+		} else {
```
Comment (Sofasmile) [inline src/Task1.java]: we don't need else block here

File: src/Main.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			} else {
+				numbers.add(scanner.nextInt());
+			}
+			index++;
+		}
+		for (index = 0; index < numbers.size(); ) {
+			stringBuilder.append(numbers.getLast()).append(" ");
+			numbers.removeLast();
+		}
+		outputNumbers = stringBuilder.toString();
```
Comment (Sofasmile) [inline src/Main.java]: we can simplify and write:
System.out.print(stringBuilder.toString());

Comment (UnitOmega13) [reply src/Main.java]: fixed

File: src/Task1.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	private static void anotherMethod() {
+		System.out.println(getCallerClassAndMethodName());
+	}
+
+	public static String getCallerClassAndMethodName() {
+		StackTraceElement[] stackTraceElement = new Exception().getStackTrace();
+		if (stackTraceElement.length >= 3) {
+			return stackTraceElement[2].getClassName() + "#" + stackTraceElement[2].getMethodName();
+		}
+			return null;
```
Comment (Sofasmile) [inline src/Task1.java]:
```suggestion
              return null;
```

File: src/Main.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import java.util.ArrayDeque;
+import java.util.Scanner;
+
+public class Main {
+
+	public static void main(String[] args) {
+		int index = 0;
+		StringBuilder stringBuilder = new StringBuilder();
+		String outputNumbers;
```
Comment (Sofasmile) [inline src/Main.java]: do we need this line here?

Comment (UnitOmega13) [reply src/Main.java]: fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**